File: CookBook.Recipes/src/CookBook.Recipes.Api/Recipes/Endpoints/SaveRecipe/Contracts/SaveRecipeRequestDto.cs
namespace CookBook.Recipes.Api.Recipes.Endpoints.SaveRecipe.Contracts;

internal sealed record SaveRecipeRequestDto
{
    public long? RecipeId { get; init; }

    public required string Title { get; init; }

    public string? Description { get; init; }

    public required short Servings { get; init; }

    public required short CookTime { get; init; }

    public string? Notes { get; init; }

    public required IReadOnlyCollection<IngredientItemDto> Ingredients { get; init; }

    public required IReadOnlyCollection<InstructionItemDto> Instructions { get; init; }

    public required IReadOnlyCollection<string> Tags { get; init; }

    public sealed record IngredientItemDto
    {
        public int? LocalId { get; init; }

        public required string Note { get; init; }
    }

    public sealed record InstructionItemDto
    {
        public int? LocalId { get; init; }

        public required string Note { get; init; }
    }
}

File: CookBook.RecipesWebapp/client/src/api/recipes/dto/SaveRecipeRequestDto.ts
export type SaveRecipeRequestDto = {
  recipeId: number | null;
  userName: string;
  title: string;
  description: string | null;
  servings: number;
  cookTime: number;
  notes: string | null;
  ingredients: IngredientItemDto[];
  instructions: InstructionItemDto[];
  tags: string[];
};

export type IngredientItemDto = {
  localId: number | null;
  note: string;
};

export type InstructionItemDto = {
  localId: number | null;
  note: string;
};
